src/apps/weblib/js-api/events.js

```diff
@@ -72,13 +81,20 @@ export default class Events {
           if (eventParam.type.endsWith('-draw') || eventParam.type.startsWith('draw-')) {
             eventParam.renderingContext = this.vizzu.render.dc()
           }
-
+        } else {
+          eventParam = param
+          eventParam.preventDefault = () => {
+            canceled = true
+          }
+        }
+        for (const handler of [...this.eventHandlers.get(eventName)[1]]) {
           handler(eventParam)
```

Comment:
This was intentional, because there is some use case, where we need to change data associated with the event. But it is true, that event objects should be immutable, so I will introduce a middleware pattern as well to support these use cases and I will leave the event objects immutable.